src/org/usfirst/frc/team2706/robot/subsystems/Bling.java

```diff
@@ -0,0 +1,147 @@
+	    else if (percent <= 0.5) bColour = colours.get("YELLOW");
+	    else if (percent <= 0.75) bColour = colours.get("PURPLE");
+	    else bColour = colours.get("GREEN");
+	    blingPort.writeString("F12C" + bColour + "D" + Math.round(percent * 100) + "E12Z"); //Use multi-colour display
+	    System.out.println("F12" + "D" + Math.round(percent * 100) + "E12Z");
+	}
+	/**
+	 * Show a distance indication on the LED strip out of 3 metres.
+	 * @param distance
+	 * : The current distance
+	 */
+	public void showDistance(double distance){
+	    blingPort.writeString("I");
+	    blingPort.writeString("E0Z"); //Clear the LED strip
+	    if (distance > 3.0) return; //Only showing 3 metres from the object.
+	    double percentDist = distance/3;
+	    System.out.println(Math.round(percentDist * pixels));
+	    String dColour;
+	    if (distance > 2) dColour = colours.get("RED");
+	    else if (distance > 1) dColour = colours.get("YELLOW");
+	    else dColour = colours.get("GREEN");
+	    System.out.println(dColour);
+	    blingPort.writeString("F7C" + dColour + "P0" + "Q" + Math.round(percentDist * pixels) + "E7Z"); //Colour flash with different colours
+	}
+	/**
+	 * This function lets you show whether or not the robot is ready
+	 * to receive a gear.
+	 * @param ready
+	 * : A boolean that indicates whether or not the robot is ready. True if yes.
+	 */
+	public void showReadyToReceiveGear(boolean ready){
+	    if (ready && !batCritical) display("Green", 3, -1, 2, 100, 0, 120); //Do not interfere with critical battery warning.
+	}
+	/**
+	 * This is used to display a basic pattern on the bling
+	 * LED lights. 
+	 * @param pattern : The type of motion or animation pattern you would like to display.
+	 * @param colour
+	 * : Colour, either as a present such as "RED", "GREEN", "WHITE" (either caps or no caps)
+	 * or in decimal format. Use a programmer calculator to determine decimal format.
+	 * <a href = "http://www.barth-dev.de/online/rgb565-color-picker/">Colour Picker</a>  
+	 * 
+	 * Presets: GREEN, RED, BLUE, YELLOW, ORANGE, PURPLE, TAN, VIOLET, MERGE, PINK, WHITE, TURQUOISE, BLACK
+	 * @param duration : The amount of seconds the whole thing lasts.
+	 * @param delay : The delay between animation segments in seconds, if applicable.
+	 * @param pixelStart : The percent of the bar where the pixel pattern will start.
+	 * @param pixelEnd : The percent of the bar where the pattern will end.
+	 */
+	public void display(String colour, int pattern, double duration, double delay, 
+	                int brightness, int pixelStart, int pixelEnd){
+	    String gColour = colour.replace(" ", ""); //Get rid of all the spaces
+	    gColour = gColour.toUpperCase(); //Make sure that any letters are uppercase.
+	    
+	    if ((gColour.charAt(0)) != '0'){ //Preset colour that we need to convert to RGB888.
+	     gColour = colours.get(gColour);
+	    }
+	   
+	    int startPixel = Math.round((pixelStart * pixels)/100);
```

Comment:
might look better to have a few spaces in between like (pixelStart * pixels) / 100);